util/dir/src/lib.rs

```diff
@@ -219,21 +253,104 @@ impl DatabaseDirectories {
 	}
 }
 
+/// Default data path
 pub fn default_data_path() -> String {
 	let app_info = AppInfo { name: PRODUCT, author: AUTHOR };
 	get_app_root(AppDataType::UserData, &app_info).map(|p| p.to_string_lossy().into_owned()).unwrap_or_else(|_| "$HOME/.parity".to_owned())
 }
 
+/// Default local path
 pub fn default_local_path() -> String {
 	let app_info = AppInfo { name: PRODUCT, author: AUTHOR };
 	get_app_root(AppDataType::UserCache, &app_info).map(|p| p.to_string_lossy().into_owned()).unwrap_or_else(|_| "$HOME/.parity".to_owned())
 }
 
+/// Default hypervisor path
 pub fn default_hypervisor_path() -> String {
 	let app_info = AppInfo { name: PRODUCT_HYPERVISOR, author: AUTHOR };
 	get_app_root(AppDataType::UserData, &app_info).map(|p| p.to_string_lossy().into_owned()).unwrap_or_else(|_| "$HOME/.parity-hypervisor".to_owned())
 }
 
+/// Get home directory.
+fn home() -> PathBuf {
+	env::home_dir().expect("Failed to get home dir")
+}
+
+/// Geth path
+pub fn geth(testnet: bool) -> PathBuf {
+	let mut base = platform::geth_base();
+	if testnet {
+		base.push("testnet");
+	}
+	base.push("keystore");
+	base
+}
+
+/// Parity path for specific chain
+pub fn parity(chain: &str) -> PathBuf {
+	let mut base = platform::parity_base();
+	base.push(chain);
+	base
+}
+
+mod platform {
+	use std::path::PathBuf;
+	#[cfg(target_os = "macos")]
```

Comment:
What I mean with `platform` module is:
```rust
// re-export
pub use platform::*;

#[cfg(target_os = "macos")]
mod platform {
  pub const CACHE_PATH: &'static str = "$BASE/cache";
  pub fn parity_base() -> PathBuf { ... }
}

#[cfg(windows)]
mod platform {
  pub const CACHE_PATH: &'static str = "$LOCAL/cache";
  pub fn parity_base() -> PathBuf { ... }
}
```
So that you don't need to repeat those `#[cfg(` statements for all methods, since it might be a bit confusing to understand what is exposed where.